model-client/src/commonMain/kotlin/org/modelix/model/client2/ReplicatedModel.kt

```diff
@@ -2,68 +2,61 @@ package org.modelix.model.client2
 
 import kotlinx.coroutines.CoroutineScope
 import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
 import kotlinx.coroutines.cancel
 import kotlinx.coroutines.delay
 import kotlinx.coroutines.launch
 import kotlinx.coroutines.sync.Mutex
 import kotlinx.coroutines.sync.withLock
+import org.modelix.model.IVersion
 import org.modelix.model.VersionMerger
 import org.modelix.model.api.IBranch
 import org.modelix.model.api.IBranchListener
+import org.modelix.model.api.IIdGenerator
 import org.modelix.model.api.ITree
 import org.modelix.model.api.PBranch
+import org.modelix.model.api.runSynchronized
 import org.modelix.model.lazy.BranchReference
 import org.modelix.model.lazy.CLTree
 import org.modelix.model.lazy.CLVersion
 import org.modelix.model.operations.OTBranch
 import org.modelix.model.server.api.ModelQuery
-import kotlin.coroutines.cancellation.CancellationException
 
-class ReplicatedModel(val client: IModelClientV2, val branchRef: BranchReference, val query: ModelQuery? = null) {
-    private val scope = CoroutineScope(Dispatchers.Default)
+class ReplicatedModel(
+    val client: IModelClientV2,
+    val branchRef: BranchReference,
+    private val providedScope: CoroutineScope? = null,
+    @Deprecated("was never supported")
+    val query: ModelQuery? = null,
```

Comment:
Is there anyone client that actually used this parameter if it didn't work? If not, we could avoid the deprecation at all and just drop it. I would assume only modelix-internal clients of this class?